mlrun/model_monitoring/api.py

```diff
@@ -39,117 +39,193 @@
-        monitoring_feature_set = mlrun.feature_store.get_feature_set(
-            uri=model_endpoint.status.monitoring_feature_set_uri
-        )
 
-        # Modify the DataFrame to the required structure that will be used later by the monitoring batch job
-        df_to_target[
-            mlrun.common.schemas.model_monitoring.EventFieldType.TIMESTAMP
-        ] = datetime.datetime.now()
-        df_to_target[
-            mlrun.common.schemas.model_monitoring.EventFieldType.ENDPOINT_ID
-        ] = endpoint_id
-        df_to_target.set_index(
-            mlrun.common.schemas.model_monitoring.EventFieldType.ENDPOINT_ID,
-            inplace=True,
-        )
+def record_results(
+    project: str,
+    model_path: str,
+    model_endpoint_name: str,
+    endpoint_id: str = "",
+    function_name: str = "",
+    context: mlrun.MLClientCtx = None,
+    df_to_target: pd.DataFrame = None,
+    sample_set_statistics: typing.Dict[str, typing.Any] = None,
+    monitoring_mode: ModelMonitoringMode = ModelMonitoringMode.enabled,
+    drift_threshold: float = 0.7,
+    possible_drift_threshold: float = 0.5,
+    trigger_monitoring_job: bool = False,
+    inf_capping: float = 10.0,
+    artifacts_tag: str = "",
+    default_batch_image="mlrun/mlrun",
+) -> ModelEndpoint:
+    """
+    Write a provided inference dataset to model endpoint parquet target. If not exist, generate a new model endpoint
+    record and use the provided sample set statistics as feature stats that will be used later for the drift analysis.
+    To manually trigger the monitoring batch job, set `trigger_monitoring_job=True` and then the batch
+    job will immediately perform drift analysis between the sample set statistics stored in the model and the new
+    input data (along with the outputs). The drift rule is the value per-feature mean of the TVD and Hellinger scores
+    according to the provided thresholds.
+
+    :param project:                  Project name.
 
-        mlrun.feature_store.ingest(
-            featureset=monitoring_feature_set, source=df_to_target, overwrite=False
+    :param model_path:               The model Store path.
+    :param model_endpoint_name:      If a new model endpoint is generated, the model endpoint name will be presented
+                                     under this endpoint.
+    :param endpoint_id:              Model endpoint unique ID. If not exist in DB, will generate a new record based
+                                     on the provided `endpoint_id`.
+    :param function_name:            If a new model endpoint is created, use this function name for generating the
+                                     function URI.
+    :param context:                  MLRun context. Note that the context is required for logging the artifacts
+                                     following the batch drift job.
+    :param df_to_target:             DataFrame that will be stored under the model endpoint parquet target. This
```

Comment:
1. any assumptions on the structure of this df? Should be stated.
2. cosider change of param name - "infer_results_df"? 
3. "DataFrame will be used by the scheduled monitoring batch drift job" - too much implementation info for user-facing funciton. how about "Will be used for doing the drift analysis"

Reply:
1. yes, it should include the inputs + output columns as expected by the batch job (corresponding to the reference data from the training process). At the moment I'll explain in the docs but in the future we can add here some kind of validation engine on that df with error handling (e.g. what to do with `None` cells). 
2. good suggestion. will change it. 
3. agreed :)